extensions/dacpac/src/wizard/api/dacFxConfigPage.ts

```diff
@@ -54,7 +54,11 @@ export abstract class DacFxConfigPage extends BasePage {
 		this.serverDropdown.onValueChanged(async () => {
 			this.model.server = (this.serverDropdown.value as ConnectionDropdownValue).connection;
 			this.model.serverName = (this.serverDropdown.value as ConnectionDropdownValue).displayName;
-			await this.populateDatabaseDropdown();
+			if (this.databaseDropdown) {
+				await this.populateDatabaseDropdown();
+			} else {
+				await this.getDatabaseValues();
```

Comment:
does setting `this.databaseNames` trigger creation of `this.databaseDropdown` or something?

Reply:
If I'm understanding the code correctly it looks like it's just that not every page has a database dropdown (for example the importConfigPage doesn't). In that case we still want to fetch the available databases so that we can do validation of the database names even if they aren't being displayed to the user. 

---
In reply to: [369288007](https://github.com/microsoft/azuredatastudio/pull/8909#discussion_r369288007) [](ancestors = 369288007)

Reply:
Yup this is so that the database names are fetched for import and deploy new db for validating the name of the new db